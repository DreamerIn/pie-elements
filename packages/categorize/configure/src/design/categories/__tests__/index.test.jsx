import { shallow } from 'enzyme';
import React from 'react';

import { Categories } from '../index';

describe('Categories', () => {
  let w;
  let onChange = jest.fn();
  let onDeleteChoice = jest.fn();
  let onAddChoice = jest.fn();
  let onAdd = jest.fn();
  let onDelete = jest.fn();
  let onColumnsChange = jest.fn();

  const wrapper = extras => {
    const defaults = {
      classes: {
        categories: 'categories',
        categoriesHolder: 'categoriesHolder',
        row: 'row'
      },
      columns: 2,
      onColumnsChange,
      categories: [{ id: '1', label: 'foo', choices: [] }],
      onDelete,
      onDeleteChoice,
      onAddChoice,
      onAdd,
      className: 'className',
      onChange
    };

    const props = { ...defaults, ...extras };
    return shallow(<Categories {...props} />);
  };

  describe('snapshot', () => {
    it('renders', () => {
      w = wrapper();
      expect(w).toMatchSnapshot();
    });
  });
  describe('logic', () => {
    describe('changeCategory', () => {
      it('calls onChange', () => {
        w = wrapper();
        const update = { id: '1', label: 'update', choices: [] };
        w.instance().changeCategory(update);
        expect(onChange).toBeCalledWith([update]);
      });
    });
  });
});
