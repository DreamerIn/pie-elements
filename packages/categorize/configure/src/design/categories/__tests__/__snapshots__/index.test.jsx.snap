// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories snapshot renders 1`] = `
<div
  className="categories className"
>
  <WithStyles(Header)
    label="Categories"
    onAdd={[MockFunction]}
  />
  <div
    className="categoriesHolder"
    style={
      Object {
        "gridTemplateColumns": "repeat(1, 1fr)",
      }
    }
  >
    <WithStyles(Category)
      category={
        Object {
          "choices": Array [],
          "id": "1",
          "label": "foo",
        }
      }
      key="0"
      onAddChoice={[MockFunction]}
      onChange={[Function]}
      onDelete={[Function]}
      onDeleteChoice={[Function]}
    />
  </div>
  <div
    className="row"
  >
    <Component
      label="Categories per row"
      max={4}
      min={1}
      onChange={[MockFunction]}
      value={2}
    />
  </div>
</div>
`;
